### PR TITLE
benchmarking: add docs

### PR DESCRIPTION
**Pull request by rwcarlsen**
Move scripts benchmark readme content into new docs system, add
clarifying prose and speedtests spec description, and add more
machine tuning detail.

### CODE REVIEW

File: framework/doc/content/documentation/application_development/performance_benchmarking.md [hunk 1]
```diff
@@ -1,39 +1,93 @@
 
-benchmark.py
--------------------
+Performance Benchmarking
```
Comment (aeslaughter) [inline framework/doc/content/documentation/application_development/performance_benchmarking.md]: Headings are `# Foo` and `## Foo`

File: framework/doc/content/documentation/application_development/performance_benchmarking.md [hunk 2]
```diff
@@ -1,39 +1,93 @@
...
+Test Harness Benchmarks
+------------------------
+
+Benchmarks can be run through the test harness (i.e.  using the `run_tests` script) by doing
+e.g. `./run_tests --run speedtests`.  When this is done, the test harness looks for test spec
+files named `speedtests` just like the `tests` files that contain regular moose test details.
+The format for these files is:
+
+```
+[Benchmarks]
```
Comment (aeslaughter) [inline framework/doc/content/documentation/application_development/performance_benchmarking.md]: Can you just include a benchmark file directly?

!listing full/path/from/root/of/repo/benchmark.i

File: framework/doc/content/documentation/application_development/performance_benchmarking.md [hunk 3]
```diff
@@ -1,39 +1,93 @@
...
+input files are well-formed and parse correctly without actually running them.
+
 
 Manual/Direct Benchmarks
-++++++++++++++++++++++++++
+--------------------------
 
-This script can also be used to manually list and directly run benchmarks without the
-test harness (for hacking, debugging, etc.).  To do this, the script reads a ``bench.list`` text
+The `[moose-repo]/scripts/benchmark.py` script can be used to manually list and directly run benchmarks without the
```
Comment (aeslaughter) [inline framework/doc/content/documentation/application_development/performance_benchmarking.md]: What is `[moose-repo]` for?

File: framework/doc/content/documentation/application_development/performance_benchmarking.md [hunk 4]
```diff
@@ -1,39 +1,93 @@
...
+
+Tuning a Benchmarking Machine
+------------------------------
 
 In order to obtain accurate results, you need to run the benchmark process(es)
 as close to isolated as possible.  On a linux system, you should e.g. use cpu
 isolation via setting kernel boot parameters:
 
- ```
+```
```
Comment (aeslaughter) [inline framework/doc/content/documentation/application_development/performance_benchmarking.md]: What is the language? bash, etc. Use "text" if you don't know.

Comment (aeslaughter) [reply framework/doc/content/documentation/application_development/performance_benchmarking.md]: You should put a language on all of the code blocks if possible, it makes the syntax highlighting better.

**Review by aeslaughter — CHANGES_REQUESTED** [review]

File: framework/doc/content/documentation/application_development/performance_benchmarking.md (hunk 3 above)
Comment (rwcarlsen) [reply framework/doc/content/documentation/application_development/performance_benchmarking.md]: their moose git repository

File: framework/doc/content/documentation/application_development/performance_benchmarking.md (hunk 4 above)
Comment (rwcarlsen) [reply framework/doc/content/documentation/application_development/performance_benchmarking.md]: text

File: framework/doc/content/documentation/application_development/performance_benchmarking.md (hunk 2 above)
Comment (rwcarlsen) [reply framework/doc/content/documentation/application_development/performance_benchmarking.md]: I'd rather have a template file that gives an overview of available options.

File: framework/doc/content/documentation/application_development/performance_benchmarking.md (hunk 3 above)
Comment (aeslaughter) [reply framework/doc/content/documentation/application_development/performance_benchmarking.md]: We usually use ~/projects/moose for this type of thing.

File: framework/doc/content/documentation/application_development/performance_benchmarking.md [hunk 5]
```diff
@@ -1,41 +1,91 @@
```
Comment (aeslaughter) [inline framework/doc/content/documentation/application_development/performance_benchmarking.md]: You need to get rid of this empty line. This is a quirk of my system that I have not yet resolved. The top level heading must be at the very top. Something I need to fix, I know the cause, just haven't got to it yet.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**